tasklist/client/src/modules/queries/useCurrentUser.tsx

```diff
@@ -15,20 +15,21 @@ function useCurrentUser() {
   return useQuery<CurrentUser, RequestError | Error>({
     queryKey: ['currentUser'],
     queryFn: async () => {
-      const {response, error} = await request(api.v1.getCurrentUser());
+      const {response, error} = await request(api.v2.getCurrentUser());
 
       if (response !== null) {
         const currentUser = await response.json();
-        const permissions: unknown[] = currentUser?.permissions ?? [];
-
+        const permissions: unknown[] | undefined = currentUser?.permissions;
```

Comment:
Why was this change necessary? The only change necessary in this file looks like it is on line 18, the rest doesn't look like it needs to be changed

```suggestion
        const permissions: unknown[] | undefined = currentUser?.permissions ?? [];
```

Reply:
I lack the reason as this was done by another engineer before I finished up the PR will revert thius + the changes further down that adopt to the `permissions` not being defined.

Reply:
it was actually because of the absence of `permissions` on the new dto, we went for removing the whole permissions logic for now